src/main/java/com/fasterxml/jackson/databind/EnumNamingStrategies.java

```diff
@@ -43,87 +135,338 @@ private EnumNamingStrategies() {}
      * <li>"Username" is converted into "username"</li>
      * </ul>
      *
-     * @since 2.15
+     * @since 2.18
      */
-    public static class CamelCaseStrategy implements EnumNamingStrategy {
+    public static class LowerCamelCaseStrategy implements EnumNamingStrategy {
 
         /**
-         * An intance of {@link CamelCaseStrategy} for reuse.
+         * An instance of {@link LowerCamelCaseStrategy} for reuse.
          *
-         * @since 2.15
+         * @since 2.18
          */
-        public static final CamelCaseStrategy INSTANCE = new CamelCaseStrategy();
+        public static final LowerCamelCaseStrategy INSTANCE = new LowerCamelCaseStrategy();
 
         /**
-         * @since 2.15
+         * @since 2.18
          */
         @Override
         public String convertEnumToExternalName(String enumName) {
-            if (enumName == null) {
-                return null;
-            }
+            return toBeanName(enumName);
+        }
+    }
 
-            final String UNDERSCORE = "_";
-            StringBuilder out = null;
-            int iterationCnt = 0;
-            int lastSeparatorIdx = -1;
-
-            do {
-                lastSeparatorIdx = indexIn(enumName, lastSeparatorIdx + 1);
-                if (lastSeparatorIdx != -1) {
-                    if (iterationCnt == 0) {
-                        out = new StringBuilder(enumName.length() + 4 * UNDERSCORE.length());
-                        out.append(toLowerCase(enumName.substring(iterationCnt, lastSeparatorIdx)));
-                    } else {
-                        out.append(normalizeWord(enumName.substring(iterationCnt, lastSeparatorIdx)));
-                    }
-                    iterationCnt = lastSeparatorIdx + UNDERSCORE.length();
-                }
-            } while (lastSeparatorIdx != -1);
+    /**
+     * <p>
+     * An implementation of {@link EnumNamingStrategy} that converts enum names in the typical upper
+     * snake case format to upper camel case format.
+     * This implementation first normalizes to lower camel case using (see {@link LowerCamelCaseStrategy} for details)
+     * and then uses {@link PropertyNamingStrategies.UpperCamelCaseStrategy} to finish converting the name.
```

Comment:
Uhhh. BAD idea. I wish I had noticed before approving -- should not plug in to internal details of another class.

Just noticed when trying to merge into 3.0 (`master`).

Reply:
I will need to revert this, actually.

Reply:
Sorry, I thought it made sense to reuse this preexisting battle-tested implementation for converting lowerCamelCase into other specific cases.

Do you have a recommendation for how this should be achieved instead? Access the same code in a different way? Implement my own logic from scratch? Copy paste?

Reply:
Right, I agree with the intent here, just not the mechanism.

Not 100% sure of a good way (can't think of a great way at least): copy pasting (with a comment pointing to source) might be ok here, given it shouldn't change a lot.
Otherwise a util class would be good. Unfortunately there isn't any obvious choice... closest might be `BeanUtil`, if method(s) in question can be static.

Reply:
... also, adding something like `NamingUtil` in `src/main/java/com/fasterxml/jackson/databind/util` would be ok if that makes more sense. It's bit dubious given only 2 use cases but... maybe that's the way to do it.

Apologies for catching this bit late.

Reply:
Here is the updated pull request:
https://github.com/FasterXML/jackson-databind/pull/4728